File: app/SantanderInvestimento/app/src/main/java/lzacheu/com/br/santanderinvestimento/MainActivity.java
package lzacheu.com.br.santanderinvestimento;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import lzacheu.com.br.santanderinvestimento.contact.ContactContract;
import lzacheu.com.br.santanderinvestimento.contact.ContactFragment;
import lzacheu.com.br.santanderinvestimento.fund.FundInfoFragment;

public class MainActivity extends AppCompatActivity {

    FragmentPageAdapter fragmentPageAdapter;

    @BindView(R.id.toolbar)
    Toolbar toolbar;

    @BindView(R.id.viewpager)
    ViewPager viewPager;

    @BindView(R.id.tabs)
    TabLayout tabLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        setSupportActionBar(toolbar);

        fragmentPageAdapter = new FragmentPageAdapter(getSupportFragmentManager());
        fragmentPageAdapter.addFragment(FundInfoFragment.newInstance(), getString(R.string.tab_investiment_title));
        fragmentPageAdapter.addFragment(ContactFragment.newInstance(), getString(R.string.tab_contact_title));
        viewPager.setAdapter(fragmentPageAdapter);
        tabLayout.setupWithViewPager(viewPager);


    }

    @Override
    protected void onResume() {
        super.onResume();

    }

    public class FragmentPageAdapter extends FragmentPagerAdapter {
        private final List<Fragment> fragmentList = new ArrayList<>();
        private final List<String> fragmentTitleList = new ArrayList<>();

        private FragmentPageAdapter(FragmentManager fm) {
            super(fm);
        }

        private void addFragment(Fragment fragment, String fragmentTitle){
            fragmentList.add(fragment);
            fragmentTitleList.add(fragmentTitle);
        }

        @Override
        public Fragment getItem(int position) {
            return fragmentList.get(position);
        }

        @Override
        public int getCount() {
            return fragmentList.size();
        }

        @Nullable
        @Override
        public CharSequence getPageTitle(int position) {
            return fragmentTitleList.get(position);
        }
    }
}

File: InvestApp/app/src/main/java/com/study/vipoliveira/investapp/ui/MainActivity.kt
package com.study.vipoliveira.investapp.ui

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v4.app.Fragment
import com.study.vipoliveira.investapp.R
import com.study.vipoliveira.investapp.ui.contact.ContactFormFragment
import com.study.vipoliveira.investapp.ui.investment.InvestFragment
import dagger.android.AndroidInjection
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.support.HasSupportFragmentInjector
import kotlinx.android.synthetic.main.activity_main.*
import javax.inject.Inject

class MainActivity : AppCompatActivity(),  HasSupportFragmentInjector{

    override fun supportFragmentInjector(): AndroidInjector<Fragment> {
        return androidInjector
    }

    @Inject lateinit var androidInjector: DispatchingAndroidInjector<Fragment>


    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val viewPagerAdapter = ViewPagerAdapter(supportFragmentManager)
        viewPagerAdapter.addFragment(InvestFragment(), getString(R.string.investment_title))
        viewPagerAdapter.addFragment(ContactFormFragment(), getString(R.string.contact_title))
        viewPager.adapter = viewPagerAdapter
        tabs.setupWithViewPager(viewPager)

    }
}
